### DISCUSSION TITLE
Using selectors not creating subscriptions to endpoints.

### DISCUSSION THREAD

**Discussion by pgsandstrom** [reduxjs/redux-toolkit#4398] [Q&A]
I have a few endpoints. Then I have some selectors that uses data from the endpoints. The real code is something similar to the examples below. What I expected was that if a component uses the selectors, it should subscribe to the endpoint in the same way as if I used the hook from the endpoint. But this isnt the case.

So my question is, what is an appropriate pattern to solve this? Should I construct hooks that wraps the selectors and also calls all the used endpoint-hooks, to ensure subscription? Im currently migrating a huge application to modern redux, so "not using selectors like this" unfortunately isnt an option right now.

selector:
```
export const useSelectMeddelande = api.endpoints.getMeddelanden.select()
```

And then in my component I have:

```
  const selectedMeddelande = useSelector(useSelectMeddelande).data
```

**Comment by EskiMojo14** [comment]
the only way to create a subscription is to use the useQuery/useLazyQuery hooks. If you only need a portion of the data, use [`selectFromResult`](https://redux-toolkit.js.org/rtk-query/usage/queries#selecting-data-from-a-query-result).

**Comment by markerikson** [comment]
As Ben said, subscriptions are done by the hooks because there's more than _just_ "select the data" - it needs to manage the subscription reference counting. The right approach is to use the appropriate query hook.
